sources/osg/Shape.js

```diff
@@ -730,6 +730,145 @@ var createGridGeometry = function ( cx, cy, cz, wx, wy, wz, hx, hy, hz, res1, re
     return g;
 };
 
+
+/*
+ * debug lines showing bbox abstraction
+ * @param col bbox color
+ */
+var createBoundingBoxGeometry = function ( col ) {
+
+    var g = new Geometry();
+    //unit cube centered on 0
+    var vertices = new Float32Array( [ -0.5, -0.5, -0.5,
+        0.5, -0.5, -0.5,
+        0.5, 0.5, -0.5, -0.5, 0.5, -0.5, -0.5, -0.5, 0.5,
+        0.5, -0.5, 0.5,
+        0.5, 0.5, 0.5, -0.5, 0.5, 0.5
+    ] );
+    g.getAttributes().Vertex = new BufferArray( BufferArray.ARRAY_BUFFER, vertices, 3 );
+
+    // use color or red
+    if ( !col ) col = [ 1.0, 0.0, 0.0, 1.0 ];
+    var colors = new MACROUTILS.Float32Array( 8 * 4 );
+    for ( var i = 0; i < 8; i++ ) {
+        for ( var k = 0; k < 4; k++ ) {
+            colors[ i * 3 + k ] = col[ k ];
+        }
+    }
+
+    g.getAttributes().Color = new BufferArray( BufferArray.ARRAY_BUFFER, colors, 4 );
+
+    var indexes = new MACROUTILS.Uint16Array(
+        [
+            //up
+            0, 1,
+            1, 2,
+            2, 3,
+            3, 0,
+            //down
+            4, 5,
+            5, 6,
+            6, 7,
+            7, 4,
+            // side
+            0, 4,
+            1, 5,
+            2, 6,
+            3, 7
+
+        ] );
+
+
+    g.getPrimitives().push( new DrawElements( PrimitiveSet.LINES, new BufferArray( 'ELEMENT_ARRAY_BUFFER', indexes, 1 ) ) );
+
+    return g;
+};
+// 
+
+/*
+ * allow for bounding boxes dynamically updated client side
```

Comment:
Rename and move updateBoundingBoxGeometryWithPerspectiveFrustum to osgUtil/debug.js

Could be named this way:

createFrustrumDebugGeometry: function( projection, dr ) {...}
You can create the debug geometry in the function and create a updateCallback that will update the geometry from the projection matrix